### PR TITLE
#10078: add option to disable csrf on forms

### PR DESCRIPTION
**Pull request by machour**

### CODE REVIEW

File: framework/helpers/BaseHtml.php [hunk 1]
```diff
@@ -300,6 +300,7 @@ public static function csrfMetaTags()
      * the attributes of the resulting tag. The values will be HTML-encoded using [[encode()]].
      * If a value is null, the corresponding attribute will not be rendered.
      * See [[renderTagAttributes()]] for details on how attributes are being rendered.
+     * The "csrf" special option can be set to FALSE to prevent the form from generating the CSRF hidden field.
```
Comment (SilverFire) [inline framework/helpers/BaseHtml.php]:
```
Special options: 
 - `csrf`: whether to generate the CSRF hidden input. When is not defined, defaults to true.
```

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -57,6 +57,7 @@ Yii Framework 2 Change Log
 - Enh #9901: Default `Cache.SerializerPermissions` configuration option for `HTMLPurifier` is set to `0775` (klimov-paul)
 - Enh #10056: Allowed any callable to be passed to `ActionColumn::$urlCreator` (freezy-sk)
 - Enh #10061: `yii\helpers\BaseInflector::transliterate()` is now public. Introduced different levels of transliteration strictness (silverfire)
+- Enh #10078: Added `csrf` option to `Html::beginForm()` to allow disabling the hidden csrf field generation. (machour)
```
Comment (SilverFire) [inline framework/CHANGELOG.md]: no need in trailing dot

**Comment by SilverFire** [comment]
Closes #10078

**Comment by SilverFire** [comment]
Looks fine as for me. 
@samdark ?

File: framework/helpers/BaseHtml.php [hunk 3]
```diff
@@ -316,7 +318,9 @@ public static function beginForm($action = '', $method = 'post', $options = [])
                 $hiddenInputs[] = static::hiddenInput($request->methodParam, $method);
                 $method = 'post';
             }
-            if ($request->enableCsrfValidation && !strcasecmp($method, 'post')) {
+            $csrf = ArrayHelper::remove($options, 'csrf', true);
+
+            if ($csrf && $request->enableCsrfValidation && !strcasecmp($method, 'post')) {
```
Comment (samdark) [inline framework/helpers/BaseHtml.php]: I'd use `strcasecmp(...) === 0`. `!` is a bit confusing in this case.

**Comment by samdark** [comment]
Looks fine except [this minor thing](https://github.com/yiisoft/yii2/pull/10182#discussion_r45182020).

File: framework/helpers/BaseHtml.php (hunk 3 above)
Comment (SilverFire) [inline framework/helpers/BaseHtml.php]: agree

**Comment by SilverFire** [comment]
Thank you!

**Pull request merged**